sensibo/unified.js

```diff
@@ -1,235 +1,485 @@
-function fanLevelToHK(value, fanLevels) {
-	if (value === 'auto') {
-		return 0
-	}
+// eslint-disable-next-line no-unused-vars
+const SensiboACPlatform = require('../sensibo/SensiboACPlatform')
+const Classes = require('../classes')
+
+module.exports = {
 
-	fanLevels = fanLevels.filter(level => {
-		return level !== 'auto'
-	})
+	/**
+	 * @param {string} value
+	 * @param {string[]} fanLevels
+	 * @returns {number}
+	 */
+	fanLevelToHK: function(value, fanLevels) {
+		if (value === 'auto') {
+			return 0
+		}
 
-	const totalLevels = fanLevels.length > 0 ? fanLevels.length : 1
-	const valueIndex = fanLevels.indexOf(value) + 1
+		fanLevels = fanLevels.filter(level => {
+			return level !== 'auto'
+		})
 
-	return Math.round(100 * valueIndex / totalLevels)
-}
+		const totalLevels = fanLevels.length > 0 ? fanLevels.length : 1
+		const valueIndex = fanLevels.indexOf(value) + 1
 
-// TODO: use Utils version instead
-function toCelsius(value) {
-	return (value - 32) / 1.8
-}
+		return Math.round(100 * valueIndex / totalLevels)
+	},
 
-// TODO: move all functions in to Utils
-module.exports = {
+	/**
+	 * Convert degrees F to degrees C
+	 * @param  {Number} degreesF The degrees in F to convert
+	 * @return {Number} The degrees in C
+	 */
+	toCelsius: function (degreesF) {
+		const degreesC = (degreesF - 32) / 1.8
+
+		return degreesC
+	},
+
+	/**
+	 * Convert degrees C to degrees F
+	 * @param  {Number} degreesC The degrees in C to convert
+	 * @return {Number} The degrees in F
+	 */
+	toFahrenheit: function (degreesC) {
+		const degreesF = Math.round((degreesC * 1.8) + 32)
+
+		return degreesF
+	},
 
-	deviceInformation: device => {
+	/**
+	 * @param {import('../types').Device} device
+	 * @returns {import('../types').DeviceInfo}
+	 */
+	getDeviceInfo: function (device) {
 		return {
 			id: device.id,
-			model: device.productModel ?? device.model,
+			productModel: device.productModel,
 			serial: device.serial,
 			manufacturer: 'Sensibo Inc.',
 			appId: 'com.sensibo.Sensibo',
-			roomName: device.room?.name ?? device.roomName,
+			room: device.room,
```

Comment:
If this is coming from the API response (and I'm pretty sure it is) then I think it should be:
```suggestion
			room: device.room.name,
```
otherwise it will return an object...

I see you worked around this by using namePrefix but I think that is unnecessary as I don't think the other fields in the room object are ever referenced